### PR TITLE
Feature/model

### PR DESCRIPTION
**Pull request by dariokozicki**
-Add initial models for products, categories, and categoryProducts (the additional model is because the joined table has more attributes -- id, created_at and updated_at)
-Add initial services for these models
-Add initial controllers for these models

### CODE REVIEW

File: src/products/models/classes/product.entity.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  @Column({ type: 'integer' })
+  rating: number;
+
+  @Column({ type: 'integer' })
+  status: number;
+
+  @Column({ type: 'integer' })
+  stock: number;
+
+  // @ManyToMany(() => Category) ANOTHER ENTITY INSTEAD OF MANY TO MANY
```
Comment (CamilaNiebles) [inline src/products/models/classes/product.entity.ts]: It is not a good practice keep code as a comment if you don't need it just, remove it (:

File: src/products/models/classes/product.entity.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { BaseEntity } from '@base/models/classes/BaseEntity.entity';
+import { CategoryProducts } from '@category_products/models/classes/category_products.entity';
+import { Column, Entity, JoinTable, OneToMany } from 'typeorm';
```
Comment (CamilaNiebles) [inline src/products/models/classes/product.entity.ts]: Also if you don't need variables as `JoinTable` you must not keep it as an import.

File: src/products/models/classes/product.entity.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { BaseEntity } from '@base/models/classes/BaseEntity.entity';
+import { CategoryProducts } from '@category_products/models/classes/category_products.entity';
```
Comment (CamilaNiebles) [inline src/products/models/classes/product.entity.ts]: Very nice approach use alias path recommendation, create a new alias path for model-classes and in this way your imports are more clean. Same in the previous line and other ones that you have.

File: src/products/tests/services/products.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { Product } from '@products/models/classes/product.entity';
+import { ProductsService } from '@products/services/products.service';
+import * as sinon from 'sinon';
```
Comment (CamilaNiebles) [inline src/products/tests/services/products.service.spec.ts]: If you are not using this dependency you must remove it.

**Review by CamilaNiebles — CHANGES_REQUESTED** [review]

File: src/products/models/classes/product.entity.ts (hunk 1 above)
Comment (dariokozicki) [reply src/products/models/classes/product.entity.ts]: Will do!

File: src/products/models/classes/product.entity.ts (hunk 3 above)
Comment (dariokozicki) [reply src/products/models/classes/product.entity.ts]: Sure thing!

**Review by CamilaNiebles — CHANGES_REQUESTED** [review]
Hey! Nice job here, now we can see layers on your implementation. Please take a look into the test, something is failing.

**Review by CamilaNiebles — APPROVED** [review]

**Pull request merged**